search/search_indexes.py

```diff
@@ -60,36 +34,57 @@ def get_course_metadata(course_id):
     Returns:
         data (dict): Metadata about course.
     """
-    check_cache_age()
-    data = INDEX_CACHE["course"].get(course_id, {})
+    key = "course_metadata_{0}".format(course_id)
+    data = cache.get(key, {})
     if data == {}:
-        course = Course.objects.get(id=course_id)
+        course = Course.objects.select_related("repository").get(id=course_id)
         data["run"] = course.run
         data["course_number"] = course.course_number
         data["org"] = course.org
         data["repo_slug"] = course.repository.slug
-        INDEX_CACHE["course"][course_id] = data
+        cache.set(key, data)
+    # Return `data` directly, not from the cache. Otherwise, if caching
+    # is disabled (TIMEOUT == 0), this will always return nothing.
     return data
 
 
-def get_vocabs(course_id, lid):
+def get_vocabs(course_id, resource_id):
```

Comment:
I don't see a good reason to get hung up on this, but this is an "API" breaking change, have you double checked all the callers and that they are just using args and not kwargs?